modules/features/settings/src/main/java/au/com/shiftyjelly/pocketcasts/settings/viewmodel/StorageSettingsViewModel.kt

```diff
@@ -158,6 +166,48 @@ class StorageSettingsViewModel
         analyticsTracker.track(AnalyticsEvent.SETTINGS_STORAGE_CLEAR_DOWNLOAD_CACHE)
     }
 
+    fun fixDownloadedFiles() {
+        analyticsTracker.track(AnalyticsEvent.SETTINGS_STORAGE_FIX_DOWNLOADED_FILES)
+        fixDownloadsJob?.cancel()
+        fixDownloadsJob = viewModelScope.launch {
+            val episodeCount = episodeManager.countEpisodes()
+            mutableState.update { state ->
+                state.copy(fixDownloadsState = State.FixDownloadsState(isFixing = true, episodeCount = episodeCount))
+            }
+
+            repeat(episodeCount / FIX_EPISODES_LIMIT + 1) { iteration ->
+                val offset = iteration * FIX_EPISODES_LIMIT
+                val episodes = episodeManager.getAllPodcastEpisodes(FIX_EPISODES_LIMIT, offset)
+                episodes.forEachIndexed { index, episode ->
+                    val fixCount = offset + index
+                    mutableState.update { state ->
+                        state.copy(fixDownloadsState = state.fixDownloadsState.copy(currentlyFixed = fixCount))
+                    }
+
+                    withContext(Dispatchers.IO) {
+                        val path = DownloadHelper.pathForEpisode(episode, fileStorage)?.takeIf {
+                            val file = File(it)
+                            file.exists() && file.isFile
+                        }
+                        if (path != null && path != episode.downloadedFilePath) {
+                            episodeManager.updateDownloadFilePath(episode, path, markAsDownloaded = true)
+                        }
+                    }
```

Comment:
Should we send a Track event here if we fix any downloads so we know if it's helping anyone?

Reply:
Maybe it would be better to send a bulk event after we're done with the number of fixed episode? What do you think?

Reply:
One event at the end with the number of fixed episodes as a parameter would be great.

Reply:
0e262e68b24b0748afa4f13b402ea33607a3eddc